pkg/proxy/ipvs/proxier.go

```diff
@@ -808,6 +808,46 @@ func (proxier *Proxier) syncProxyRules() {
 	// activeBindAddrs represents ip address successfully bind to DefaultDummyDevice in this round of sync
 	activeBindAddrs := map[string]bool{}
 
+	hasNodePort := false
+	for _, svc := range proxier.serviceMap {
+		svcInfo, ok := svc.(*serviceInfo)
+		if ok && svcInfo.NodePort() != 0 {
+			hasNodePort = true
+			break
+		}
+	}
+
+	// Both nodeAddresses and nodeIPs can be reused for all nodePort services
+	// and only need to be computed if we have at least one nodePort service.
+	var (
+		// List of node addresses to listen on if a nodePort is set.
+		nodeAddresses []string
+		// List of node IP addresses to be used as IPVS services if nodePort is
+		// set.
+		nodeIPs []net.IP
+	)
+
+	if hasNodePort {
+		nodeAddrSet, err := utilproxy.GetNodeAddresses(proxier.nodePortAddresses, proxier.networkInterfacer)
+		if err != nil {
+			klog.Errorf("Failed to get node ip address matching nodeport cidr: %v", err)
+		}
+		if nodeAddrSet.Len() > 0 {
```

Comment:
This will panic if err != nil above

Reply:
Actually this works because the set implementation will simply call len() on the underlying map which will return 0 even for a nil map: https://play.golang.org/p/ZO4U9-VvxPs. I agree this is confusing though, I'll update this condition to `if err == nil && nodeAddrSet.Len() > 0`

Reply:
Good point though on the nil map

Reply:
Good point though on the nil map